cpp/src/replace/clamp.cu

```diff
@@ -123,12 +123,12 @@ std::unique_ptr<cudf::column> clamp_string_column(strings_column_view const& inp
       size_type idx) {
       if (d_input.is_null(idx)) { return; }
       auto input_element      = d_input.element<string_view>(idx);
-      const auto d_lo         = (*lo_itr).first;
-      const auto d_hi         = (*hi_itr).first;
-      const auto d_lo_replace = (*lo_replace_itr).first;
-      const auto d_hi_replace = (*hi_replace_itr).first;
-      const auto lo_valid     = (*lo_itr).second;
-      const auto hi_valid     = (*hi_itr).second;
+      const auto d_lo         = *(*lo_itr);
```

Comment:
What does this return if `(*lo_itr).has_value()` is false?

Reply:
It will be undefined value in those cases. I see your point and I can update the algorithm to only conditionally de-reference the `lo_itr` and `hi_itr` when they have a value.

Reply:
Looking at this more, I think we can use `value_or`